rttest/src/rttest.cpp

```diff
@@ -741,9 +740,15 @@ int Rttest::lock_and_prefault_dynamic()
 
 int rttest_prefault_stack_size(const size_t stack_size)
 {
+  /*
   unsigned char stack[stack_size];
   memset(stack, 0, stack_size);
   return 0;
+  */
+  unsigned char * stack = new unsigned char[stack_size];
+  memset(stack, 0, stack_size);
+  delete[] stack;
+  return 0;
```

Comment:
So it turns out that these two things aren't equivalent.  The code that was there was a stack allocation, using VLA (which is what I guess the warning was about).  The new code is a heap allocation, so it doesn't actually prefault the stack anymore.

What's the warning message you were seeing with this in place?

Reply:
Here's the warning:
```
/root/ros2_ws/src/ros2/realtime_support/rttest/src/rttest.cpp:743:17: warning: ISO C++ forbids variable length array ‘stack’ [-Wvla]
  743 |   unsigned char stack[stack_size];
      |  
```

Reply:
Right, VLAs.  You can instead try:

```
unsigned char *stack = alloca(stack_size);
```

Which is also a stack allocation.  Is that any happier?

Reply:
Yup, seems to work (after a `static_cast`).